frontend/src/old-pages/Clusters/Cost.tsx

```diff
@@ -0,0 +1,293 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+// with the License. A copy of the License is located at
+//
+// http://aws.amazon.com/apache2.0/
+//
+// or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+// OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+// limitations under the License.
+import React from 'react';
+import { useState, consoleDomain, setState, clearState } from '../../store'
+// import { getIn } from '../../util'
+// import { useCollection } from '@awsui/collection-hooks';
+
+
+// UI Elements
+import {
+  Button,
+  Box,
+  BarChart,
+  DateRangePicker,
+  Container,
+  Header,
+  SpaceBetween,
+} from "@awsui/components-react";
+
+// Components
+import HelpTooltip from '../../components/HelpTooltip'
+import { ActivateTags, GetTagStatus, GetGraphData } from '../../model';
+import { Star } from '@mui/icons-material';
+import { totalmem } from 'os';
+
+export default function Cost() {
+  const cluster_name = useState(['app', 'clusters', 'selected']);
+  const tag_active = useState(['app', 'tags', 'active']);
+  const y_domain = useState(['app', 'cost', cluster_name, 'max']);
+  const [dateValue, setDateValue] = React.useState(undefined);
+  const dates = useState(['app', 'cost', cluster_name, 'dates']) || [];
+  const stack_info = useState(['app', 'cost', cluster_name, 'stack_info']) || [];
+
+  // Check tag status
+  const check_tags = () => {
+    const status_update = (value: {TagStatus?: boolean} ) => {
+      setState(['app', 'tags', 'active'], value['TagStatus']);
+    }
+    GetTagStatus(status_update)
+  }
+
+  // Activate all the tags
+  const activate_tags = () => {
+    const status_update = () => {
+      setState(['app', 'tags', 'active'], true);
+    }
+    ActivateTags(status_update)
+  }
+
+  // Pull cost and usage from CE API
+  const get_graph_data = (Start: string, End: string) => {
+    const query_costs = (data: any) => {
+      const usage_data = data['ResultsByTime'];
+      const dates: Date[] = [];
+      const instance_types = new Set();
+      const stack_info: any[] = [];
+      let day_info: any[] = [];
+      let total = 0;
+      let max = 0;
+  
+      // Find XDomain
+      const x_domain = usage_data?.map((obj: any) => {
```

Comment:
obj is a bit generic, is it possible to come up with a more descriptive name here?